apps/backend/src/libs/plugins/method-guard/method-guard.plugin.ts

```
@@ -0,0 +1,56 @@
+import { type FastifyPluginCallback } from "fastify";
+import fp from "fastify-plugin";
+
+import { ExceptionMessage, ServerErrorType } from "~/libs/enums/enums.js";
+import { HTTPCode } from "~/libs/modules/http/http.js";
+
+import { matchRoute } from "./libs/match-route.js";
+
+type Options = {
+	allRoutes: { method: string; path: string }[];
+};
+
+const methodGuardCallback: FastifyPluginCallback<Options> = (
+	fastify,
+	options,
+	done,
+) => {
+	fastify.addHook("onRequest", (request, response, next) => {
+		const { allRoutes } = options;
+		const method = request.method.toUpperCase();
+		const [url] = response.request.url.split("?");
+
+		if (!url) {
+			return response.status(HTTPCode.NOT_FOUND).send({
+				errorType: ServerErrorType.COMMON,
+				message: ExceptionMessage.ROUTE_NOT_FOUND,
+			});
+		}
```

Can we throw an error instead? Then the error will caught by error handler